src/test/java/org/mockitousage/debugging/InvocationListenerCallbackTest.java

```diff
@@ -39,6 +46,37 @@ public void should_call_single_listener_when_mock_return_normally() throws Excep
     }
 
     @Test
+    public void should_call_listeners_in_order() throws Exception {
+        // given
+        List<InvocationListener> container = new ArrayList<InvocationListener>();
+        RememberingListener listener1 = new RememberingListener(container);
+        RememberingListener listener2 = new RememberingListener(container);
+
+        Foo foo = mock(Foo.class, withSettings().invocationListeners(listener1, listener2));
+
+        // when
+        foo.giveMeSomeString("herb");
+
+        // then
+        assertThat(container).containsExactly(listener1, listener2);
+    }
+
+    @Test
+    public void should_allow_same_listener() throws Exception {
+        // given
+        List<InvocationListener> container = new ArrayList<InvocationListener>();
+        RememberingListener listener1 = new RememberingListener(container);
+
+        Foo foo = mock(Foo.class, withSettings().invocationListeners(listener1, listener1));
+
+        // when
+        foo.giveMeSomeString("herb");
+
+        // then
+        assertThat(container).containsExactly(listener1, listener1);
```

Comment:
@szczepiq What do you thing about a test to document behavior how many times the listener is called if provided multiple times in `invocationListeners`?

Reply:
I can add it. Good idea to document and preserve the current behavior of the public API.